django/contrib/sitemaps/__init__.py

```diff
@@ -122,6 +124,14 @@ def location(self, item):
 
     def get_protocol(self, protocol=None):
         # Determine protocol
+        if not any((self.protocol, protocol)):
+            warnings.warn(
+                "The default sitemap protocol will be changed from 'http' to "
+                "'https' in Django 5.0. Explicitly pass 'protocol' to this "
+                "method or to the constructor to silence this warning.",
```

Comment:
I don't think that users call `get_protocol()` directly. I would rather recommend to set the `Sitemap.protocol` attribute (which is documented), e.g.
```suggestion
                "The default sitemap protocol will be changed from 'http' to "
                "'https' in Django 5.0. Set Sitemap.protocol to 'http' if you "
                "want to keep the current default behavior.",
```